src/main/java/tutorspet/logic/commands/student/EditStudentCommand.java

```diff
@@ -78,6 +78,11 @@ public CommandResult execute(Model model) throws CommandException {
         Student studentToEdit = lastShownList.get(index.getZeroBased());
         Student editedStudent = createEditedStudent(studentToEdit, editStudentDescriptor);
 
+        // Filter list to get a list without the studentToEdit, then check if the editedStudent already exist in list.
+        if (lastShownList.stream().filter(s -> !s.equals(studentToEdit)).anyMatch(editedStudent::isSameStudent)) {
```

Comment:
We can't use `lastShownList` because it does not contain all students in Tutor's Pet. The model has to handle this check.

Reply:
Can consider overloading the `hasStudent` method in `TutorsPet` and I think the guard clause below can be removed.